packages/contracts/test/manager/facets/ManagerFacet.t.sol

```diff
@@ -2,101 +2,191 @@
     }
 
     function testShouldSetBondingShareAddress() public prankAs(admin) {
-        ManagerFacet(address(diamond)).setBondingShareAddress(contract1);
-        assertEq(ManagerFacet(address(diamond)).getBondingShareAddress(), contract1);
+        IManagerFacet.setBondingShareAddress(contract1);
+        assertEq(IManagerFacet.getBondingShareAddress(), contract1);
     }
 
     function testShouldSetStableSwapMetaPoolAddress() public prankAs(admin) {
-        ManagerFacet(address(diamond)).setStableSwapMetaPoolAddress(contract1);
-        assertEq(ManagerFacet(address(diamond)).getStableSwapMetaPoolAddress(), contract1);
+        IManagerFacet.setStableSwapMetaPoolAddress(contract1);
+        assertEq(IManagerFacet.getStableSwapMetaPoolAddress(), contract1);
     }
 
     function testShouldSetBondingContractAddress() public prankAs(admin) {
-        ManagerFacet(address(diamond)).setBondingContractAddress(contract1);
-        assertEq(ManagerFacet(address(diamond)).getBondingContractAddress(), contract1);
+        IManagerFacet.setBondingContractAddress(contract1);
+        assertEq(IManagerFacet.getBondingContractAddress(), contract1);
     }
 
     function testShouldSetTreasuryAddress() public prankAs(admin) {
-        ManagerFacet(address(diamond)).setTreasuryAddress(contract1);
-        assertEq(ManagerFacet(address(diamond)).getTreasuryAddress(), contract1);
+        IManagerFacet.setTreasuryAddress(contract1);
+        assertEq(IManagerFacet.getTreasuryAddress(), contract1);
     }
 
     function testShouldSetIncentiveToUAD() public prankAs(admin) {
         address dollarTokenAddress = generateAddress("dollarTokenAddress", true, 10 ether);
-        ManagerFacet(address(diamond)).setDollarTokenAddress(dollarTokenAddress);
-        ManagerFacet(address(diamond)).setIncentiveToUAD(user1, contract1);
+        IManagerFacet.setDollarTokenAddress(dollarTokenAddress);
+        IManagerFacet.setIncentiveToUAD(user1, contract1);
+    }
+
+    function testShouldSetMinterRoleWhenInitializing() public prankAs(admin) {
+        assertEq(IManagerFacet.hasRole(UBQ_MINTER_ROLE, admin), true);
+    }
+
+    function testShouldInitializeDollarTokenAddress() public prankAs(admin) {
+        assertEq(IManagerFacet.getDollarTokenAddress(), address(diamond));
+    }
+
+    function testShouldDeployStableSwapPool() public {
+
+        vm.startPrank(admin);
+
+        MockuADToken uAD;
+        MockUbiquityGovernance uGov;
+
+        uAD = new MockuADToken(10000);
+        uGov = new MockUbiquityGovernance(10000);
+
+        IManagerFacet.setDollarTokenAddress(address(uAD));
+        IManagerFacet.setGovernanceTokenAddress(address(uGov));
+        IERC20 crvToken = IERC20(0x6c3F90f043a72FA612cbac8115EE7e52BDe6E490);
+
+        address secondAccount = address(0x3);
+        address bondingZeroAccount = address(0x4);
+        address bondingMinAccount = address(0x5);
+        address bondingMaxAccount = address(0x6);
```

Comment:
I'm going to let the variable names ask slide because I realize that theres a lot of renames that would be needed. @ryzhak perhaps I could burden you with a quick pass on this according to the spec I provided you the other week. Sunny has already been working on this PR for a long time and I think we're all ready to merge it in already.